File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
// m3hu1
class Solution {
public:
    static bool cmp(pair<int, int>& a,pair<int, int>& b) 
    { 
        return a.second > b.second; 
    } 
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> ans;
        unordered_map<int,int> mp;
        for(int i = 0 ; i < nums.size(); i++){
            mp[nums[i]]++;
        }
        vector<pair<int,int>> A;
        for(auto it : mp){
            A.push_back(it);
        }
        sort(A.begin(),A.end(),cmp);
        int count = 0;
        for(auto it : A){
            if(count == k){
                break;
            }
            else {
                ans.push_back(it.first);
                count++;
            }
        }
        return ans;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ans = []
        count = Counter(nums)
        freq = [[] for _ in range(len(nums) + 1)]

        for i, j in count.items():
            freq[j].append(i)
        
        for i in range(len(freq) - 1, -1, -1):
            for val in freq[i]:
                ans.append(val)
                if len(ans) == k:
                    return ans
        
        return []